### PR TITLE
handbrake: 1.5.1 -> 1.6.1

### LINKED ISSUE #211297: handbrake: cant build its ffmpeg
### Describe the bug
As title says, the ffmpeg that handbrake uses can't be built, complaining about missing sdl2.

### Steps To Reproduce
1. Try to install handbrake
2. Get the error

### Expected behavior
Handbrake is installable

### Additional context
```
error: builder for '/nix/store/l0gbll3kck6jr8f61affw0ja88znv4jk-ffmpeg-full-4.4.1.drv' failed with exit code 1;
       last 10 log lines:
       > patching file libavcodec/hevc_mp4toannexb_bsf.c
       > configuring
       > configure flags: --disable-static --prefix=/nix/store/kq9iks048pqjwvbzajvcgnsj057wva0z-ffmpeg-full-4.4.1 --target_os=linux --arch=x86_64 --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --disable-lto --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --enable-ffmpeg --enable-ffplay --enable-ffprobe --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avutil --enable-postproc --enable-swresample --enable-swscale --enable-doc --disable-htmlpages --enable-manpages --disable-podpages --disable-txtpages --enable-bzlib --enable-libcelt --enable-cuda --enable-cuda-llvm --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdc1394 --enable-libdrm --enable-iconv --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmysofa --enable-libopenmpt --enable-libopus --enable-librsvg --enable-libsrt --enable-libssh --enable-libtheora --enable-libv4l2 --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-libxml2 --enable-lzma --enable-cuvid --enable-nvdec --enable-nvenc --enable-openal --enable-opencl --enable-libopencore-amrnb --disable-opengl --enable-libopenh264 --enable-libopenjpeg --disable-openssl --enable-libpulse --enable-librav1e --enable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --enable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-libzimg --enable-zlib --enable-vulkan --enable-libglslang --enable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
       > ERROR: sdl2 requested but not found
       >
       > If you think configure made a mistake, make sure you are using the latest
       > version from Git.  If the latest version fails, report the problem to the
       > ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
       > Include the log file "ffbuild/config.log" produced by configure as this will help
       > solve the problem.
       For full logs, run 'nix log /nix/store/l0gbll3kck6jr8f61affw0ja88znv4jk-ffmpeg-full-4.4.1.drv'.
error: 1 dependencies of derivation '/nix/store/yl2qnvs5i7i7l0xsms62mcim24r7pckr-handbrake-1.5.1.drv' failed to build
building '/nix/store/78k0bh5malzd32dm77nkcm56xs3m3q37-discord-0.0.22.drv'...
error: 1 dependencies of derivation '/nix/store/5b7pszn77b5n66chnl2zp6a2ikzz5q8z-handbrake-1.5.1-fish-completions.drv' failed to build

```

### Notify maintainers
@acowley @wmertens @Anton-Latukha 
### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.3-zen1, NixOS, 23.05 (Stoat), 23.05pre443927.6dccdc45851`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos"`
 - channels(m911t): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

### PR DESCRIPTION
**Pull request by laalsaas**
Fixes #211297

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/handbrake/default.nix [hunk 1]
```diff
@@ -25,7 +25,7 @@
 , numactl
 , writeText
   # Processing, video codecs, containers
-, ffmpeg-full
+, ffmpeg_4
```
Comment (QuantMint) [inline pkgs/applications/video/handbrake/default.nix:28]: Why don't you use ffmpeg5 instead, handbrake [supports it]("https://github.com/HandBrake/HandBrake-contribs/releases).

Comment (laalsaas) [reply pkgs/applications/video/handbrake/default.nix]: I just went towards updating handbrake alltogeher.

**Comment by uninsane** [comment]
this runs fine for me on x86_64-linux (wayland/sway), but the new svt-avt package input added in this patch breaks the aarch64-linux build in a new way:

```console
$ nix build '.#legacyPackages.aarch64-linux.handbrake'
error: Package ‘svt-av1-1.4.1’ in /nix/store/1w97dryirwh60407sg97x521m0aisln4-source/pkgs/tools/video/svt-av1/default.nix:31 is marked as broken, refusing to evaluate.
```

ffmpeg-full special-cases a few dependencies, presumably to work around platform-specific issues like this:
```nix
ffmpeg-full = callPackage ../development/libraries/ffmpeg-full {
    svt-av1 = if stdenv.isAarch64 then null else svt-av1;
    rtmpdump = null; # Prefer the built-in RTMP implementation
    # The following need to be fixed on Darwin
    libjack2 = if stdenv.isDarwin then null else libjack2;
    libmodplug = if stdenv.isDarwin then null else libmodplug;
    libmfx = if stdenv.isDarwin then null else intel-media-sdk;
    libpulseaudio = if stdenv.isDarwin then null else libpulseaudio;
    samba = if stdenv.isDarwin then null else samba;
    inherit (darwin.apple_sdk.frameworks)
      Cocoa CoreServices CoreAudio AVFoundation MediaToolbox
      VideoDecodeAcceleration VideoToolbox;
  };
  ```
  
  i guess if you want to ditch it maybe you have to replicate some of that?

File: pkgs/applications/video/handbrake/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/applications/video/handbrake/default.nix]: If v1.6.1 by default uses ffmpeg5, so should we.

**Comment by laalsaas** [comment]
The problem here is that svt-av1 is a dependency of handbrake, not ffmpeg. So setting svt-av1 to `null` would result in a failing build on aarch64. I haven't been able to figure out if there is a way to build handbrake 1.6.1 without svt-av1.

File: pkgs/applications/video/handbrake/default.nix [hunk 2]
```diff
@@ -84,58 +85,57 @@
...
+      "${src}/contrib/ffmpeg/A22-matroskaenc-aac-extradata-updated.patch"
+      "${src}/contrib/ffmpeg/A23-ccaption_dec-fix-pts-in-real_time-mode.patch"
+      "${src}/contrib/ffmpeg/A24-fix-eac3-dowmix.patch"
+      "${src}/contrib/ffmpeg/A25-enable-truehd-pass.patch"
+      "${src}/contrib/ffmpeg/A26-Update-the-min-version-to-1.4.23.0-for-AMF-SDK.patch"
+      "${src}/contrib/ffmpeg/A27-avcodec-amfenc-Fixes-the-color-information-in-the-ou.patch"
+      "${src}/contrib/ffmpeg/A28-avcodec-amfenc-HDR-metadata.patch"
+      "${src}/contrib/ffmpeg/A30-svt-av1-backports.patch"
     ];
+    #configureFlags = lib.remove "--enable-sdl2" old.configureFlags;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/handbrake/default.nix:138]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
If the package now requires svt-avt and that does not work on aarch64-linux that is unfortunate but I don't think we can do anything here unless we find some patch upstream or ask them for help.

**Comment by laalsaas** [comment]
I agree with Sandro here.

@uninsane svt-av1 was marked as broken 2 years ago, can you try if a new version fixed the issue on aarch64 and nixpkgs just forgot to set broken to false? I don't have aarch64 hardware, so i can't test.

**Comment by uninsane** [comment]
it’ll be 3-4 days before i get a chance to do that, but yeah if nobody beats me to it i’ll look into the svt-av1 stuff 👍

you might also try cross-compiling: you can probably tell a lot here without actually deploying/running it on HW. handbrake with broken-at-runtime AV1 on aarch64 would still be much better than handbrake uncompilable for aarch64.

**Comment by QuantMint** [comment]
> I agree with Sandro here.
> 
> @uninsane svt-av1 was marked as broken 2 years ago, can you try if a new version fixed the issue on aarch64 and nixpkgs just forgot to set broken to false? I don't have aarch64 hardware, so i can't test.

Cross compiling stv-av1 for aarch64 works just fine.
```
❯ export NIXPKGS_ALLOW_BROKEN=1
❯ nix build nixos\#pkgsCross.aarch64-multiplatform.svt-av1 --impure --print-out-paths
/nix/store/n0kvp195xn02brcsbq7ga3y0cnavwbfh-svt-av1-aarch64-unknown-linux-gnu-1.4.1
```
It think that you can just remove this: 
https://github.com/NixOS/nixpkgs/blob/e1e1b192c1a5aab2960bf0a0bd53a2e8124fa18e/pkgs/tools/video/svt-av1/default.nix#L46

**Comment by SuperSandro2000** [comment]
darwin aarch64 is still broken but that might be related to the apple SDK.

File: pkgs/tools/video/svt-av1/default.nix [hunk 3]
```diff
@@ -43,6 +43,5 @@ stdenv.mkDerivation (finalAttrs: {
     license = with licenses; [ aom bsd3 ];
     maintainers = with maintainers; [ Madouura ];
     platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/video/svt-av1/default.nix:45]:
```suggestion
    platforms = platforms.unix;
    # error: use of undeclared identifier 'kCVPixelFormatType_444YpCbCr16BiPlanarVideoRange'
    broken = stdenv.isAarch64 && stdenv.isDarwin;
```

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**